src/Azure.Functions.Cli/Actions/AzureActions/LogStreamAction.cs

```diff
@@ -42,5 +68,33 @@ public override async Task RunAsync()
                 }
             }
         }
+
+        public async Task OpenLiveStreamInBrowser(Site functionApp, ArmSubscriptionsArray allSubscriptions)
+        {
+            if (!functionApp.AzureAppSettings.ContainsKey(ApplicationInsightsIKeySetting))
+            {
+                throw new CliException($"Missing {ApplicationInsightsIKeySetting} App Setting. " +
+                    $"Please make sure you have Application Insights configured with you function app.");
+            }
+
+            var iKey = functionApp.AzureAppSettings[ApplicationInsightsIKeySetting];
+            if (string.IsNullOrEmpty(iKey))
+            {
+                throw new CliException("Invalid Instrumentation Key found. Please make sure that the Application Insights is configured correctly.");
+            }
+
+            var appId = await AzureHelper.GetApplicationInsightIDFromIKey(iKey, AccessToken, ManagementURL, allSubs: allSubscriptions);
```

Comment:
what happens if the application insights resource doesn't exist?

Reply:
The function `GetApplicationInsightIDFromIKey` takes care of it and throws the right error - complaining about the instrumentation key.

If key doesn't exist, that is taken care of above.

Reply:
sounds good.